zebra-network/src/peer/handshake.rs

```diff
@@ -376,22 +384,44 @@ where
                 future::ready(Ok(msg))
             });
 
+            // CORRECTNESS
+            //
+            // Every message and error must update the peer address state via
+            // the inbound_ts_collector.
+            let inbound_ts_collector = timestamp_collector.clone();
             let peer_rx = peer_rx
                 .then(move |msg| {
-                    // Add a metric for inbound messages and fire a timestamp event.
-                    let mut timestamp_collector = timestamp_collector.clone();
+                    // Add a metric for inbound messages and errors.
+                    // Fire a timestamp or failure event.
+                    let mut inbound_ts_collector = inbound_ts_collector.clone();
                     async move {
-                        if let Ok(msg) = &msg {
-                            metrics::counter!(
-                                "zcash.net.in.messages",
-                                1,
-                                "command" => msg.to_string(),
-                                "addr" => addr.to_string(),
-                            );
-                            use futures::sink::SinkExt;
-                            let _ = timestamp_collector
-                                .send(MetaAddr::new_responded(&addr, &remote_services))
-                                .await;
+                        match &msg {
+                            Ok(msg) => {
+                                metrics::counter!(
+                                    "zcash.net.in.messages",
+                                    1,
+                                    "command" => msg.to_string(),
+                                    "addr" => addr.to_string(),
+                                );
+                                use futures::sink::SinkExt;
```

Comment:
It seems to me that none of the `SinkExt` imports (there are 3 of them in this file) are actually needed. Unsure why we have no warning about them (maybe because they are in `async` blocks?) but i tried removing them all and all the tests complete fine. I think it will be fine to remove them in this PR.

Reply:
SinkExt is included in `futures::prelude::*`:
https://docs.rs/futures/0.3.13/futures/prelude/index.html

Reply:
```suggestion
```

Reply:
Thanks for the clarification.